### PR TITLE
Separate data uploading from job submission for DS2 cloud training and add support for multiple shards uploading.

### PR DESCRIPTION
**Pull request by xinghai-sun**
Resolve #205 

- Separate data uploading from training job submission for DS2 cloud training.
- Add supports for multiple shards packing and uploading.
- Update cloud/REAME.md

### CODE REVIEW

File: deep_speech_2/cloud/README.md [hunk 1]
```diff
@@ -1,81 +1,70 @@
...
+
+-  `IN_MANIFESTS`： Paths (in local filesystem) of manifest files containing the audio files to be uploaded. Multiple paths can be concatenated with a whitespace delimeter. Lines of manifest files are in the following format:
 
 ```
 {"audio_filepath": "/home/disk1/LibriSpeech/dev-clean/1272/128104/1272-128104-0000.flac", "duration": 5.855, "text
 ": "mister quilter is the ..."}
 {"audio_filepath": "/home/disk1/LibriSpeech/dev-clean/1272/128104/1272-128104-0001.flac", "duration": 4.815, "text
 ": "nor is mister ..."}
 ```
+- `OUT_MANIFESTS`: Paths (in local filesystem) to write the updated output manifest files to. Multiple paths can be concatenated with a whitespace delimeter. The values of `audio_filepath` in the output manifests are jjjjjkknew paths in PaddleCloud filesystem.
```
Comment (wanghaoshuang) [inline deep_speech_2/cloud/README.md]: jjjjjkknew  -> new ?

File: deep_speech_2/cloud/README.md [hunk 2]
```diff
@@ -1,81 +1,70 @@
...
+By running:
 
->Note: Upload will be skipped if target file has existed in `CLOUD_DATA_DIR`.
+```
+sh pcloud_upload_data.sh
+```
+all the audio files will be uploaded to PaddleCloud filesystem, and you will get modified manifests files in `OUT_MANIFESTS`.
 
-## Step-2  Configure computation resource
+You have to take this step only once, when it is your first time to do the cloud training. Later on, the data is persisitent on the cloud filesystem and is reusable for multple jobs.
```
Comment (wanghaoshuang) [inline deep_speech_2/cloud/README.md]: multple -> multiple

File: deep_speech_2/cloud/pcloud_submit.sh [hunk 3]
```diff
@@ -1,48 +1,27 @@
-# Configure input data set in local filesystem
-TRAIN_MANIFEST="../datasets/manifest.train"
-DEV_MANIFEST="../datasets/manifest.dev"
-VOCAB_FILE="../datasets/vocab/eng_vocab.txt"
-MEAN_STD_FILE="../mean_std.npz"
-# Configure output path in PaddleCloud filesystem
-CLOUD_DATA_DIR="/pfs/dlnel/home/sunxinghai@baidu.com/deepspeech2/data"
+TRAIN_MANIFEST="cloud/cloud.manifest.test"
```
Comment (wanghaoshuang) [inline deep_speech_2/cloud/pcloud_submit.sh]: 'cloud.manifest.test' -> 'cloud.manifest.train'?

File: deep_speech_2/cloud/upload_data.py [hunk 4]
```diff
@@ -22,66 +19,81 @@
...
 parser.add_argument(
-    "--train_manifest_path",
-    default="../datasets/manifest.train",
-    type=str,
-    help="Manifest file path for train data. (default: %(default)s)")
-parser.add_argument(
-    "--dev_manifest_path",
-    default="../datasets/manifest.dev",
+    "--in_manifest_paths",
+    default=["../datasets/manifest.test", "../datasets/manifest.dev"],
```
Comment (wanghaoshuang) [inline deep_speech_2/cloud/upload_data.py]: 这里默认值是故意不设置为`/manifest.train`么？

File: deep_speech_2/cloud/upload_data.py [hunk 5]
```diff
@@ -99,44 +111,12 @@ def pcloud_cp(src, dst):
...
-    pcloud_cp(local_dev_tar, cloud_dev_tar)
+    if not os.path.exists(args.local_tmp_dir):
+        os.makedirs(args.local_tmp_dir)
+    pcloud_mkdir(args.cloud_data_dir)
 
-    # upload vocab file and mean_std file
-    pcloud_cp(args.vocab_file, cloud_vocab_file)
-    pcloud_cp(args.mean_std_file, cloud_mean_file)
+    upload_data(args.in_manifest_paths, args.out_manifest_paths,
+                args.local_tmp_dir, args.cloud_data_dir, 10)
```
Comment (wanghaoshuang) [inline deep_speech_2/cloud/upload_data.py]: 10 -> args.num_shards

**Review by wanghaoshuang — CHANGES_REQUESTED** [review]

File: deep_speech_2/cloud/README.md (hunk 1 above)
Comment (xinghai-sun) [reply deep_speech_2/cloud/README.md]: Done.

File: deep_speech_2/cloud/README.md (hunk 2 above)
Comment (xinghai-sun) [reply deep_speech_2/cloud/README.md]: Done.

File: deep_speech_2/cloud/pcloud_submit.sh (hunk 3 above)
Comment (xinghai-sun) [reply deep_speech_2/cloud/pcloud_submit.sh]: Done.

File: deep_speech_2/cloud/upload_data.py (hunk 4 above)
Comment (xinghai-sun) [reply deep_speech_2/cloud/upload_data.py]: 改过来了，不是故意的，调试时临时设置忘记改了。

File: deep_speech_2/cloud/upload_data.py (hunk 5 above)
Comment (xinghai-sun) [reply deep_speech_2/cloud/upload_data.py]: Done.

**Review by wanghaoshuang — APPROVED** [review]

**Pull request merged**